### ISSUE TITLE
[@babel/parser@7.12.1]: unexpected `SyntaxError` omitted from function params destructuring

### ISSUE THREAD

**Issue by shhider** [babel/babel#12195]
## Bug Report

Common object deconstruction code:

```typescript
interface ISomeObj {
  prop1: string;
  prop2: string;
}

const clone = (
  { prop1, prop2 }: ISomeObj,
): ISomeObj => ({ prop1, prop2 });
```

but babel/parser@7.12.1:

```
/repl.ts: Unexpected token (8:0)

  6 | const clone = (
  7 |   { prop1, prop2 }: ISomeObj,
> 8 | ): ISomeObj => ({ prop1, prop2 });
    | ^
  9 | 
```

You can try it in babel website `Try It Out`.

And,

There is no error if param has only one prop:

```typescript
interface ISomeObj {
  prop1: string;
  prop2: string;
}

const clone = (
  { prop1 }: ISomeObj,
): ISomeObj => ({ prop1 });
```

This bug blocks my development, please fix it ASAP.

**Comment by personalZhang** [comment]
+1

**Comment by nicolo-ribaudo** [comment]
I reverted the commit that introduced the regression and released it as `@babel/parser@7.12.2`.

**Comment by shhider** [comment]
Appreciate!
